### DISCUSSION TITLE
Maintain maximized window size when switching between various size monitors?

### DISCUSSION THREAD

**Discussion by JamIsJam88** [rxhanson/Rectangle#645] [Q&A]
Is there a way to keep windows maximized when switching between monitors of different resolutions?

I have a Macbook Pro 2021 14 inch, which is not quite 1920x1080 or even the same aspect ratio as my two other monitors. When switching windows between the Macbook and either of the two monitors, the windows reduce in size to the resolution/aspect ratio of the Macbook. Therefore, I always have to use the keyboard shortcut to maximize the windows after they go on the Macbook. It also happens when I close an app that was on the Macbook screen and reopen it. It's a minor issue but slightly annoying.

**Comment by rxhanson** [comment]
Right now, the behavior you are looking for will only happen if your window was maximized by Rectangle before moving it across displays. I'll look into making this happen regardless of how your window became maximized for a future release.

**Reply by JamIsJam88** [reply]
Hey, thanks for responding. Yeah, it works great when you open the window and maximize it with Rectangle. However, if you close the app and reopen it, the window may open maximized on that monitor, but it won't stay maximized when moving it across displays. I was wondering if I was missing a setting. It would be amazing if windows stayed maximized even after reopening the app and switching displays.
